lesson17/index.js

```diff
@@ -0,0 +1,171 @@
+    const button = document.createElement("div");
+    button.classList.add("sliderBtn", `${sliderDirection}`);
+    button.id = `js-slider-${sliderDirection}Btn`;
+    button.setAttribute("data-btntype", `${sliderDirection}`)
+    sliderDirection === "previous" && button.classList.add("is-disabled");
+    main.querySelector(".slider").appendChild(button);
+  });
+  setBtnEvent();
+}
+
+const controlBtnBehavior = () => {
+  const allSlides = Array.from(document.querySelectorAll(".sliderImg"));
+  const allSlidesLength = Array.from(document.querySelectorAll(".sliderImg")).length;
+  const currentSlide = document.querySelector('[data-hidden="false"]');
+  const prevBtn = document.getElementById("js-slider-previousBtn");
+  const nextBtn = document.getElementById("js-slider-nextBtn");
+  const currentSlideIndex = allSlides.findIndex((slide)=>{
+    return slide === currentSlide;
+  });
+
+  if (currentSlideIndex === 0) {
+    prevBtn.classList.add("is-disabled");
+    nextBtn.classList.remove("is-disabled");
+  } else if(currentSlideIndex === (allSlidesLength - 1)) {
+    prevBtn.classList.remove("is-disabled");
+    nextBtn.classList.add("is-disabled");
+  } else {
+    prevBtn.classList.remove("is-disabled");
+    nextBtn.classList.remove("is-disabled");
+  }
+}
+
+const setFraction = (slides) => {
+  const slider = document.getElementById("js-slider");
+  const fraction = document.createElement("div");
+  fraction.classList.add("fraction");
+  const fractionWrapper = document.createElement("div");
+
+  const numerator = document.createElement("span");
+  numerator.id = "js-numerator";
+  numerator.textContent = "1";
+  const separator = document.createElement("span");
+  separator.textContent = "/";
+  const denominator = document.createElement("span");
+  denominator.textContent = slides.length;
+
+  fractionWrapper.appendChild(numerator);
+  fractionWrapper.appendChild(separator);
+  fractionWrapper.appendChild(denominator);
+
+  fraction.appendChild(fractionWrapper);
+  slider.appendChild(fraction);
+}
+
+const updateFractionNum = ()=> {
+  const allSlides = Array.from(document.querySelectorAll(".sliderImg"));
+  const currentSlide = document.querySelector('[data-hidden="false"]');
+  const currentSlideIndex = allSlides.findIndex((slide)=>{
+    return slide === currentSlide;
+  })
+  const numerator = document.getElementById("js-numerator");
+  const currentSlideNum = currentSlideIndex + 1;
+  numerator.textContent = currentSlideNum;
+}
+
+const setLoadingImage = () => {
+  const fragmentLoadingImage = document.createDocumentFragment();
+  const div = document.createElement("div");
+  div.id = "js-loading";
+  const img = document.createElement("img");
+  img.src = "./loading-circle.gif";
+  fragmentLoadingImage.appendChild(div).appendChild(img);
+  main.appendChild(fragmentLoadingImage);
+}
+
+const removeLoadingImage = () => {
+  const loadingImage = document.getElementById("js-loading");
+  main.removeChild(loadingImage);
+}
```

Comment:
IMO
It may be a good idea to just process the erasing instead of removing it from the parent element.
```suggestion
const removeLoadingImage = () => {
  document.getElementById("js-loading").remove();
}
```